### PR TITLE
chore: Added updating of docs site with compat table

### LINKED ISSUE #2157: Automate the generation of compatibility report
### Description
On release of the agent it should PR the docs-website with a new compatibility report.  The workflow should also allow to run manually at anytime.

### Acceptance Criteria
On release of the agent I want the most up to date support versions of a library to show on `docs-website`.  I also want the ability to generate report and PR docs-website at any time.

### Additional context
The process lives [here](https://github.com/newrelic/newrelic-node-versions).  The release should publish as a go package so the agent can install this package and run the script.  The workflow should also PR the `docs-website`

### PR DESCRIPTION
**Pull request by jsumners-nr**
This PR should resolve #2157.

### CODE REVIEW

File: .github/workflows/ci-workflow.yml
Comment (jsumners-nr) [inline .github/workflows/ci-workflow.yml:1]: I simplified this workflow while I was at it (mainly to fix it so that I didn't have to wait on all the runs while working on this PR). Turns out, I had a typo that was breaking things (`output` should be `outputs`). But when I really reviewed it, I realized that the "skip release" was really trying to accomplish the same task. So I refactored things a bit and now we should only need the one conditional check (if it works as I intend).

File: .github/workflows/compatibility-report.yml [hunk 1]
```diff
@@ -54,17 +54,59 @@ jobs:
...
         with:
+          token: ${{ secrets.NODE_AGENT_GH_TOKEN || secrets.GITHUB_TOKEN }}
           title: "docs: Updated compatibility report"
           commit-message: "docs: Updated compatibility report"
           branch: "compatibility-report/auto-update"
           delete-branch: true
           base: main
           labels: "documentation"
 
+  docs:
```
Comment (jsumners-nr) [inline .github/workflows/compatibility-report.yml:69]: I'm not yet sure this job will do what we actually want, but I can't run it from a branch because this PR is originating from a fork. I think it will just be easier to merge it as-is, and then I can do some manual testing and submit another PR if things are not correct.

File: .github/workflows/ci-workflow.yml [hunk 2]
```diff
@@ -12,39 +12,25 @@ jobs:
...
-      javascript_changed: ${{ steps.filter.output.javascript }}
+      javascript_changed: ${{ steps.filter.outputs.javascript }}
     steps:
       - uses: actions/checkout@v4
       - uses: dorny/paths-filter@v3
         id: filter
         with:
           filters: |
             javascript:
-              - '**/*.js'
```
Comment (bizob2828) [inline .github/workflows/ci-workflow.yml:23]: verify nice

File: .github/workflows/compatibility-report.yml [hunk 3]
```diff
@@ -54,17 +54,59 @@ jobs:
           path: compatibility.md
 
       # Generate the new PR to update the doc in the repo.
-      - run: |
```
Comment (bizob2828) [inline .github/workflows/compatibility-report.yml:57]: why was this removed?

File: .github/workflows/compatibility-report.yml [hunk 4]
```diff
@@ -54,17 +54,59 @@ jobs:
...
-          git config user.name $GITHUB_ACTOR
-          git config user.email gh-actions-${GITHUB_ACTOR}@github.com
       - run: |
           rm -f status.log
       - uses: peter-evans/create-pull-request@6d6857d36972b65feb161a90e484f2984215f83e
         with:
+          token: ${{ secrets.NODE_AGENT_GH_TOKEN || secrets.GITHUB_TOKEN }}
```
Comment (bizob2828) [inline .github/workflows/compatibility-report.yml:61]: not sure i following why you're checking an or here. shouldn't we always use our bot token which is `NODE_AGENT_GH_TOKEN`?

File: .github/workflows/compatibility-report.yml (hunk 1 above)
Comment (bizob2828) [reply .github/workflows/compatibility-report.yml:69]: that's fair

File: .github/workflows/compatibility-report.yml (hunk 3 above)
Comment (jsumners-nr) [reply .github/workflows/compatibility-report.yml:57]: The `create-pull-request` action actually does this internally. I discovered it while reviewing the documentation for this PR.

File: .github/workflows/compatibility-report.yml (hunk 4 above)
Comment (jsumners-nr) [reply .github/workflows/compatibility-report.yml:61]: As a guard for forks. I'm really not sure what happens if it is run from a fork that doesn't have our secrets.

File: .github/workflows/ci-workflow.yml (hunk 2 above)
Comment (bizob2828) [reply .github/workflows/ci-workflow.yml:23]: it just occurred to me that if we only update a dep this CI won't run. but that would also cause the prep release PR to run because the package.json will change as well so maybe we need both?

**Review by bizob2828 — APPROVED** [review]
I can't test this but LGTM
